warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PullupUnexecutableNodesVisitor.java

```diff
@@ -152,61 +145,79 @@ private boolean containsChild(JexlNode node, ExecutableDeterminationVisitor.STAT
 
     @Override
     public Object visit(ASTAndNode node, Object data) {
-        // we need to make sure one of these nodes executable, and we have no partials
-        // TODO we should use some cost/index stats info
-
-        // determine if we have any executable children
-        if (!ExecutableDeterminationVisitor.isExecutable(node, config, indexedFields, indexOnlyFields, nonEventFields, forFieldIndex, null, helper)) {
+        // if a delayed predicate, then change it to a regular reference
+        QueryPropertyMarker.Instance instance = QueryPropertyMarker.findInstance(node);
+        if (instance.isType(DELAYED)) {
+            JexlNode source = QueryPropertyMarker.unwrapFully(node, DELAYED);
 
-            boolean executable = containsChild(node, ExecutableDeterminationVisitor.STATE.EXECUTABLE);
+            JexlNode parent = (node.jjtGetParent() instanceof ASTReferenceExpression) ? node.jjtGetParent() : node;
+            source = (source instanceof ASTReferenceExpression) ? source.jjtGetChild(0) : source;
 
-            // TODO: this will basically eliminate any work we have done based on cost to delayed
-            // contained predicates. However cost got us into a non-executable situation in the first
-            // place so now we are looking for anything we can do to make this query executable.
+            // when pulling up a delayed marker that is negated, a reference expression must be persisted
+            source = wrapIfNeeded(source);
 
-            // first flip the error states regardless of the executable state
-            for (int i = 0; i < node.jjtGetNumChildren(); i++) {
-                JexlNode child = node.jjtGetChild(i);
-                ExecutableDeterminationVisitor.STATE state = ExecutableDeterminationVisitor.getState(child, config, indexedFields, indexOnlyFields,
-                                nonEventFields, forFieldIndex, null, helper);
-                if (state == ExecutableDeterminationVisitor.STATE.ERROR) {
-                    child.jjtAccept(this, data);
-                    state = ExecutableDeterminationVisitor.getState(child, config, indexedFields, indexOnlyFields, nonEventFields, forFieldIndex, null, helper);
-                }
-                if (state == ExecutableDeterminationVisitor.STATE.EXECUTABLE) {
-                    executable = true;
+            swap(parent.jjtGetParent(), parent, source);
+            return source;
+        } else if (!ExecutableDeterminationVisitor.isExecutable(node, config, indexedFields, indexOnlyFields, nonEventFields, forFieldIndex, null, helper)) {
+            // we need to make sure one of these nodes executable, and we have no partials
+            // TODO we should use some cost/index stats info
+
+            // determine if we have any executable children
+            if (!ExecutableDeterminationVisitor.isExecutable(node, config, indexedFields, indexOnlyFields, nonEventFields, forFieldIndex, null, helper)) {
```

Comment:
no need to call this method again here.

Reply:
Sure, I can make that change.  This was a byproduct of moving the reference node logic into the and node, so we were running this check redundantly in the previous code for what it's worth.  Tests pass with this change.